### ISSUE TITLE
inl_hash_join doesn't work

### ISSUE THREAD

**Issue by XuHuaiyu** [pingcap/tidb#15489]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. What did you do?
``` sql
CREATE TABLE `t` ( `a` bigint(20) NOT NULL, `b` tinyint(1) DEFAULT NULL, `c` datetime DEFAULT NULL, `d` int(10) unsigned DEFAULT NULL, `e` varchar(20) DEFAULT NULL, `f` double DEFAULT NULL, `g` decimal(30,5) DEFAULT NULL, `h` float DEFAULT NULL, `i` date DEFAULT NULL, `j` timestamp NULL DEFAULT NULL, PRIMARY KEY (`a`), UNIQUE KEY `b` (`b`), KEY `c` (`c`,`d`,`e`), KEY `f` (`f`), KEY `g` (`g`,`h`), KEY `g_2` (`g`), UNIQUE KEY `g_3` (`g`), KEY `i` (`i`) );

tidb> desc select /*+ inl_hash_join(t2) */ * from t t1 left join t t2 on t2.a=t1.a right join t t3 on t3.a=t2.a;
```

### 2. What did you expect to see?
index hash join can be used

### 3. What did you see instead?
merge join is used, and an inapplicable warning is raised

### 4. What version of TiDB are you using? (`tidb-server -V` or run `select tidb_version();` on TiDB)
0c5e58107dad2fbf5ae3464d9b9719ee503cc03a in master

**Comment by eurekaka** [comment]
This problem has already been fixed in latest master(a97e886a12c67969a2a6c818222e62b3cb4ddf1b) and 4.0 branch(c38f1887277c4d554eb9bf3f4e6fe5e51a821e43), so close it.

**Comment by ChenPeng2013** [comment]
verified on
master: v4.0.0-beta.2-632-g6c305e2a4
release-4.0: v4.0.0-66-gcf781a347
